### PR TITLE
Introduce shortlinks

### LINKED ISSUE #478: Link shortener
It would be nice to have an option to define shortlinks that are redirecting to other URLs.

### PR DESCRIPTION
**Pull request by janno42**
fix #478

### CODE REVIEW

File: _1327/main/utils.py [hunk 1]
```diff
@@ -64,3 +72,21 @@ def find_root_menu_items(items):
 
 	real_root_items.extend(find_root_menu_items(questionable_root_items))
 	return real_root_items
+
+
+def slugify_and_clean_url_title(instance, url_title):
+	from _1327.documents.models import Document
+	from _1327.shortlinks.models import Shortlink
+
+	if not slugify(url_title) == url_title:
```
Comment (Bartzi) [inline _1327/main/utils.py]: please check this call, it will break for instance if a url_title ends with `/`

File: _1327/shortlinks/tests.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+class TestShortlinkWeb(WebTest):
+
+	def setUp(self):
+		self.user = mommy.make(UserProfile)
+		self.document = mommy.make(InformationDocument, text="Internal shortlink example")
+		assign_perm(InformationDocument.VIEW_PERMISSION_NAME, self.user, self.document)
+
+	def test_follow_shortlink_external(self):
+		url = "http://github.com"
```
Comment (Bartzi) [inline _1327/shortlinks/tests.py]: why not `https://`?

File: _1327/shortlinks/views.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+from django.shortcuts import get_object_or_404, Http404, redirect, render
+from django.utils import timezone
+from django.utils.translation import ugettext_lazy as _
+
+from .forms import ShortlinkForm
+from .models import Shortlink
+
+
+def shortlinks_index(request):
+	if not request.user.is_superuser:
```
Comment (Bartzi) [inline _1327/shortlinks/views.py]: only superusers are allowed to create/edit shortlinks?
Is this really the intention?

Comment (janno42) [inline _1327/shortlinks/views.py]: yes

File: _1327/main/utils.py (hunk 1 above)
Comment (janno42) [inline _1327/main/utils.py]: fixed

File: _1327/shortlinks/tests.py (hunk 2 above)
Comment (janno42) [inline _1327/shortlinks/tests.py]: fixed

**Pull request merged**